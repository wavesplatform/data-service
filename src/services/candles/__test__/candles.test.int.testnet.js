const { of: taskOf } = require('folktale/concurrency/task');

const { createPgDriver } = require('../../../db');
const { loadConfig } = require('../../../loadConfig');

const options = loadConfig();
const create = require('..').default;

const amountAsset = '8dzLYRNtYR6ASG2W4h3FqeeY49paRxNheQwRW6CpP1HT';
const priceAsset = '4zjSCagDvgPkTCwFjvE4KMFtXz1WX4dNaNutyBw8XnrG';

describe('Candles', () => {
  const service = create({
    drivers: { pg: createPgDriver(options) },
    emitEvent: () => () => null,
    validatePair: () => taskOf(undefined),
  });

  describe('search all candles between 2017-05-26 00:00:00 and 2017-05-26 23:59:59', () => {
    it('should return all Candles correctly for each 1h', done => {
      service
        .search({
          amountAsset,
          priceAsset,
          timeStart: new Date('2017-05-26T00:00:00.000Z'),
          timeEnd: new Date('2017-05-26T23:59:59.999Z'),
          interval: '1h',
          matcher: options.matcher.defaultMatcherAddress,
        })
        .run()
        .listen({
          onResolved: value => {
            expect(value).toMatchSnapshot();
            done();
          },
          onRejected: done.fail,
        });
    });

    it('should return all Candles correctly for each 1d', done => {
      service
        .search({
          amountAsset,
          priceAsset,
          timeStart: new Date('2017-05-26T00:00:00.000Z'),
          timeEnd: new Date('2017-05-26T23:59:59.999Z'),
          interval: '1d',
          matcher: options.matcher.defaultMatcherAddress,
        })
        .run()
        .listen({
          onResolved: value => {
            expect(value).toMatchSnapshot();
            done();
          },
          onRejected: done.fail,
        });
    });
  });
});
