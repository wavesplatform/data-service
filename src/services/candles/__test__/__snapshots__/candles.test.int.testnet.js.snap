// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Candles search all candles between 2017-05-26 00:00:00 and 2017-05-26 23:59:59 should return all Candles correctly for each 1d 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "candle",
      "data": Object {
        "close": "1200",
        "high": "1200",
        "low": "1200",
        "maxHeight": 91358,
        "open": "1200",
        "quoteVolume": "43.2",
        "time": 2017-05-26T21:00:00.000Z,
        "txsCount": 36,
        "volume": "0.03600339",
        "weightedAveragePrice": "1200",
      },
    },
  ],
}
`;

exports[`Candles search all candles between 2017-05-26 00:00:00 and 2017-05-26 23:59:59 should return all Candles correctly for each 1h 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-25T21:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-25T22:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-25T23:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T00:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T01:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T02:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T03:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T04:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T05:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T06:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T07:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T08:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": "1200",
        "high": "1200",
        "low": "1200",
        "maxHeight": 90345,
        "open": "1200",
        "quoteVolume": "12724.33",
        "time": 2017-05-26T12:00:00.000Z,
        "txsCount": 604,
        "volume": "10.60361187",
        "weightedAveragePrice": "1200",
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T10:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T11:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T12:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T13:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T14:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T15:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T16:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T17:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T18:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T19:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
    Object {
      "__type": "candle",
      "data": Object {
        "close": null,
        "high": null,
        "low": null,
        "maxHeight": null,
        "open": null,
        "quoteVolume": null,
        "time": 2017-05-26T20:00:00.000Z,
        "txsCount": null,
        "volume": null,
        "weightedAveragePrice": null,
      },
    },
  ],
}
`;
